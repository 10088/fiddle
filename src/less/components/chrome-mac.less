.darwin {
  .chrome {
    text-align: center;
    background-color: @background-1;
    border-bottom: @border;
    -webkit-app-region: drag;
    height: 23px;
    font-size: 1.4rem;

    small,
    p,
    span {
      user-select: none;
    }
  }
}
