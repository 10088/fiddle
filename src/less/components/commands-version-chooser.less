button.version-chooser.bp3-button {
  border-radius: 3px;
}
