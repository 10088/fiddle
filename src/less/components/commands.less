@import "../variables.less";

header {
  box-shadow: 0 0 0 1px rgba(16, 22, 26, 0.2), 0 0 0 rgba(16, 22, 26, 0), 0 1px 1px rgba(16, 22, 26, 0.4);
  background-color: @background-1;
  -webkit-app-region: drag;
}

.commands {
  display: flex;
  justify-content: space-between;
  height: 50px;
  padding: 10px;
  box-sizing: border-box;
  font-family: @fonts-common;

  button,
  input,
  select {
    -webkit-app-region: no-drag;
    margin-bottom: 0;
  }

  select {
    width: 100px;
    margin-left: 5px;
  }

  & > div {
    display: flex;

    &:last-of-type {
      margin-right: 10px;

      button {
        margin-left: 5px;
      }
    }
  }

  .address-bar {
    height: 28px;
    width: 360px;
    transition: width 0.4s;
    display: flex;

    input,
    span {
      font-family: @fonts-common;
      font-weight: 300;
      font-size: 11px;
      cursor: auto;
      -webkit-app-region: no-drag;
    }

    input {
      flex-grow: 1;
      background: white;
      padding-left: 1rem;
      height: 2.6rem;
    }

    input:invalid {
      box-shadow: inset 0 0 0 1px #e4252d;
    }

    span {
      background: white;
      color: #1d1d1d;
      border-bottom-color: rgb(209, 209, 209);
      border-bottom-style: solid;
      border-bottom-left-radius: 4px;
      border-bottom-width: 1px;
      border-image-outset: 0;
      border-image-repeat: stretch;
      border-image-slice: 100%;
      border-image-source: none;
      border-image-width: 1;
      border-left-color: rgb(209, 209, 209);
      border-left-style: solid;
      border-left-width: 1px;
      border-top-color: rgb(209, 209, 209);
      border-top-left-radius: 4px;
      border-top-style: solid;
      border-top-width: 1px;
      display: inline-block;
      height: 1.2rem;
      letter-spacing: normal;
      margin: 0;
      padding: 6px 0 10px 10px;
      user-select: text;
      width: 132px;
      text-rendering: auto;
      text-shadow: none;
      text-transform: none;
    }

    &.empty {
      width: 200px;
    }

    &:focus-within {
      span,
      input {
        border-color: @foreground-2;
      }
    }
  }
}
