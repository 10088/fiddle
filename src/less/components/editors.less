.mosaic-tile {
  animation: fadein 0.15s;
}

.mosaic-window-toolbar  {
  & > div {
    display: flex;
    justify-content: space-between;
    width: 100%;
    padding: 0 3px 0 29px;

    .mosaic-controls {
      transform: scale(0.7);
    }

    h5 {
      margin: 4px 0 0 0;
    }

    button {
      margin-left: 8px;
    }
  }
}

@keyframes fadein {
  from { opacity: 0; }
  to { opacity: 1; }
}
