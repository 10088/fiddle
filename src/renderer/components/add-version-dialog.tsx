import { observer } from 'mobx-react';
import * as path from 'path';
import * as React from 'react';
import * as semver from 'semver';

import { NpmVersion } from '../../interfaces';
import { getElectronNameForPlatform } from '../../utils/electron-name';
import { AppState } from '../state';
import { Dialog } from './dialog';

export interface AddVersionDialogProps {
  appState: AppState;
}

export interface AddVersionDialogState {
  isValidElectron: boolean;
  isValidVersion: boolean;
  file?: File;
  version: string;
}

/**
 * The "add version" dialog allows users to add custom builds of Electron.
 *
 * @class AddVersionDialog
 * @extends {React.Component<AddVersionDialogProps, AddVersionDialogState>}
 */
@observer
export class AddVersionDialog extends React.Component<AddVersionDialogProps, AddVersionDialogState> {
  constructor(props: AddVersionDialogProps) {
    super(props);

    this.state = {
      isValidVersion: false,
      isValidElectron: false,
      version: ''
    };

    this.onSubmit = this.onSubmit.bind(this);
    this.onClose = this.onClose.bind(this);
    this.onChangeFile = this.onChangeFile.bind(this);
    this.onChangeVersion = this.onChangeVersion.bind(this);
  }

  /**
   * Handles a change of the file input
   *
   * @param {React.ChangeEvent<HTMLInputElement>} event
   */
  public async onChangeFile(event: React.ChangeEvent<HTMLInputElement>) {
    const { files } = event.target;
    const { binaryManager } = this.props.appState;
    const file = files && files[0] ? files[0] : undefined;

    const isValidElectron = !!(file
      && file.path
      && await binaryManager.getIsDownloaded('custom', file.path));

    this.setState({ file, isValidElectron });
  }

  public onChangeVersion(event: React.ChangeEvent<HTMLInputElement>) {
    const version = event.target.value || '';
    const isValidVersion = !!semver.valid(version);

    this.setState({
      version,
      isValidVersion,
     });
  }

  /**
   * Handles the submission of the dialog
   *
   * @returns {Promise<void>}
   */
  public async onSubmit(): Promise<void> {
    const { file, version } = this.state;

    if (!file) return;

    const name = file.path
      .slice(-20)
      .split(path.sep)
      .slice(1)
      .join(path.sep);

    const toAdd: NpmVersion = {
      localPath: file.path,
      version,
      name
    };

    this.props.appState.addLocalVersion(toAdd);
    this.onClose();
  }

  /**
   * Closes the dialog
   */
  public onClose() {
    this.props.appState.isAddVersionDialogShowing = false;
    this.reset();
  }

  /**
   * Reset this component's state
   */
  public reset(): void {
    this.setState({
      isValidElectron: false,
      isValidVersion: false,
      version: '',
      file: undefined
    });
  }

  get buttons() {
    const canSubmit = this.state.isValidElectron && this.state.isValidVersion;

    return [
      (
        <button
          className='button'
          key='submit'
          disabled={!canSubmit}
          onClick={this.onSubmit}
        >
          Add
        </button>
      ), (
        <button
          className='cancel'
          key='cancel'
          onClick={this.onClose}
        >
          Cancel
        </button>
      )
    ];
  }

  public render() {
    const { isAddVersionDialogShowing } = this.props.appState;
    const dirOptions = { webkitdirectory: 'true' };

    return (
      <Dialog
        isShowing={isAddVersionDialogShowing}
        isShowingBackdrop={true}
        buttons={this.buttons}
        onClose={this.onClose}
        isCentered={true}
        className='add-version-dialog'
        key='add-version-dialog'
      >
        <label
          htmlFor='custom-electron-version'
          className='force-button'
        >
          Select the folder containing {getElectronNameForPlatform()}
        </label>
        <input
          type='file'
          onChange={this.onChangeFile}
          id='custom-electron-version'
          name='custom-electron-version'
          {...dirOptions}
        />
        {this.renderPath()}
      </Dialog>
    );
  }

  private renderPath(): JSX.Element | null {
    const { file, isValidElectron } = this.state;

    if (!file || !file.path) return null;

    const info = isValidElectron
      ? `We found an ${getElectronNameForPlatform()} in this folder.`
      : `We did not find a ${getElectronNameForPlatform()} in this folder...`;

    return (
      <>
        <input
          readOnly={true}
          value={file.path}
        />
        <p>
          {info}
        </p>
        {this.renderVersionInput()}
      </>
    );
  }

  private renderVersionInput(): JSX.Element | null {
    const { isValidElectron, isValidVersion, version } = this.state;

    if (!isValidElectron) return null;

    return (
      <>
        <p>
          Please specify a version, used for typings and the name.
          Must be <code>semver</code> compliant.
        </p>
        <input
          className={isValidVersion ? '' : 'hasError'}
          value={version}
          onChange={this.onChangeVersion}
          placeholder='4.0.0'
        />
      </>
    );
  }

}
