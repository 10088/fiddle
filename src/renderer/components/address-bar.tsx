import * as React from 'react';
import { observer } from 'mobx-react';
import * as Octokit from '@octokit/rest';
import * as classNames from 'classnames';

import { AppState } from '../state';
import { INDEX_HTML_NAME, MAIN_JS_NAME, RENDERER_JS_NAME } from '../constants';
import { idFromUrl } from '../../utils/gist';
import { reaction } from 'mobx';

export interface AddressBarProps {
  appState: AppState;
}

export interface AddressBarState {
  value: string;
}

@observer
export class AddressBar extends React.Component<AddressBarProps, AddressBarState> {
  constructor(props: AddressBarProps) {
    super(props);

    this.loadFiddle = this.loadFiddle.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);

    this.state = {
      value: this.props.appState.gistId
    };
  }

  /**
   * Handle the form's submit event, trying to load whatever
   * URL was entered.
   *
   * @param {React.SyntheticEvent<HTMLFormElement>} event
   * @memberof AddressBar
   */
  public async handleSubmit(event: React.SyntheticEvent<HTMLFormElement>) {
    event.preventDefault();

    this.props.appState.gistId = idFromUrl(this.state.value) || this.state.value;
    if (this.state.value) {
      this.loadFiddle();
    }
  }

  /**
   * Once the component mounts, we'll subscribe to gistId changes
   */
  public componentDidMount() {
    reaction(
      () => this.props.appState.gistId,
      (gistId: string) => this.setState({ value: gistId })
    );
  }

  /**
   * Handle the change event, which usually just updates the address bar's value
   *
   * @param {React.ChangeEvent<HTMLInputElement>} event
   * @memberof AddressBar
   */
  public handleChange(event: React.ChangeEvent<HTMLInputElement>) {
    this.setState({ value: idFromUrl(event.target.value) || event.target.value });
  }

  /**
   * Load a fiddle
   *
   * @returns
   * @memberof AddressBar
   */
  public async loadFiddle() {
    const { gistId } = this.props.appState;

    if (!confirm('Are you sure you want to load a new fiddle, all current progress will be lost?')) return;

    try {
      const octo = new Octokit();
      const gist = await octo.gists.get({
        gist_id: gistId,
        id: gistId,
      });

      window.ElectronFiddle.app.setValues({
        html: gist.data.files[INDEX_HTML_NAME].content,
        main: gist.data.files[MAIN_JS_NAME].content,
        renderer: gist.data.files[RENDERER_JS_NAME].content,
      });
    } catch (error) {
      console.warn(`Loading fiddle failed`, error);
    }
  }

  public render() {
    const { isUnsaved } = this.props.appState;
    const { value } = this.state;
    const className = classNames('address-bar', isUnsaved, { empty: !value });

    return (
      <form className={className} onSubmit={this.handleSubmit}>
        <span>https://gist.github.com/</span>
        <input
          key='addressbar'
          type='text'
          placeholder='...'
          value={value}
          onChange={this.handleChange}
        />
      </form>
    );
  }
}
