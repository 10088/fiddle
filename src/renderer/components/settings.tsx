import { faTimesCircle } from '@fortawesome/fontawesome-free-solid';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { observer } from 'mobx-react';
import * as React from 'react';

import { classNames } from '../../utils/classnames';
import { AppState } from '../state';
import { CreditsSettings } from './settings-credits';
import { ElectronSettings } from './settings-electron';
import { GeneralSettings } from './settings-general';

enum SettingsSections {
  General = 'General',
  Electron = 'Electron',
  Credits = 'Credits'
}

const settingsSections = [
  SettingsSections.General,
  SettingsSections.Electron,
  SettingsSections.Credits
];

export interface SettingsProps {
  appState: AppState;
}

export interface SettingsState {
  section: SettingsSections;
}

/**
 * Everything above the editors, so buttons and the address bar.
 *
 * @class Settings
 * @extends {React.Component<SettingsProps, SettingsState>}
 */
@observer
export class Settings extends React.Component<SettingsProps, SettingsState> {
  constructor(props: SettingsProps) {
    super(props);

    this.state = {
      section: SettingsSections.General
    };
  }

  /**
   * Renders the content of the settings, usually by simply
   * return the appropriate component.
   *
   * @returns {(JSX.Element | null)}
   */
  public renderContent(): JSX.Element | null {
    const { section } = this.state;
    const { appState } = this.props;

    if (section === SettingsSections.General) {
      return <GeneralSettings appState={appState} />;
    }

    if (section === SettingsSections.Electron) {
      return <ElectronSettings appState={appState} />;
    }

    if (section === SettingsSections.Credits) {
      return <CreditsSettings appState={appState} />;
    }

    return null;
  }

  /**
   * Renders the individual menu items
   *
   * @returns {Array<JSX.Element>}
   */
  public renderOptions(): Array<JSX.Element> {
    const { section } = this.state;

    return settingsSections.map((name) => {
      const isSelected = section === name;
      const className = classNames({ selected: isSelected }, name);
      const onClick = () => this.setState({ section: name });

      return (
        <li onClick={onClick} key={name} className={className}>{name}</li>
      );
    });
  }

  public render() {
    const { appState } = this.props;
    const { isSettingsShowing } = appState;

    if (!isSettingsShowing) return null;

    return (
      <div className='settings'>
        <div className='settings-menu'>
          <ul>{this.renderOptions()}</ul>
        </div>
        <div className='settings-content'>
          <div className='settings-close' onClick={appState.toggleSettings}>
            <FontAwesomeIcon icon={faTimesCircle} />
          </div>
          {this.renderContent()}
        </div>
      </div>
    );
  }
}
