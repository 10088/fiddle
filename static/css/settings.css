.settings {
  background: var(--electron-background-2);
  z-index: 10;
  position: absolute;
  height: 100%;
  width: 100vw;
  display: flex;
  user-select: none;
  cursor: initial;
}

.settings h1,
.settings h2,
.settings h3,
.settings h4,
.settings h5 {
  color: white;
  cursor: initial;
}

.settings label {
  font-weight: 200;
}

.settings-content {
  height: 100%;
  width: 100%;
  padding: 20px;
  overflow: scroll;
  color: rgb(230, 230, 230);
}

.settings-menu {
  height: 100%;
  width: 250px;
  background-color: rgba(0, 0, 0, 0.25);
  user-select: none;
}

.settings-menu ul {
  list-style: none;
  padding: 20px;
}

.settings-menu li {
  padding-bottom: 5px;
  border-bottom: solid 1px rgba(255, 255, 255, 0.2);
  color: rgba(255, 255, 255, .8);
}

.settings-menu li.selected {
  color: var(--electron-bright);
}

.settings-menu li:hover {
  padding-bottom: 5px;
  border-bottom: solid 1px rgba(255, 255, 255, 0.2);
  color: rgb(255, 255, 255);
  cursor: pointer;
}

.settings-section {
  margin-bottom: 20px;
}

.settings-section:last-of-type {
  margin: 0 0 0 0;
}

.settings .electron-versions-table button {
  margin: 0;
  min-width: 250px;
}

.settings .electron-versions-table .action {
  text-align: center;
  width: 250px;
}

.settings .contributors {
  display: flex;
  flex-wrap: wrap;
}

.settings .contributor {
  width: 320px;
  height: 120px;
  background: rgba(255, 255, 255, .1);
  padding: 10px;
  margin: 0 10px 10px 0;
  display: flex;
  justify-content: flex-start;
  cursor: pointer;
}

.settings .contributor:hover {
  background: rgba(255, 255, 255, .15);
}

.settings .contributor .avatar {
  height: 100px;
  width: 100px;
  background-size: contain;
  margin-right: 10px;
}

.settings .contributor h5,
.settings .contributor p {
  margin-bottom: 0;
}

.settings .contributor .details {
  max-width: 190px;
  overflow: scroll;
}

.settings .settings-close {
  position: fixed;
  right: 30px;
  height: 25px;
  width: 25px;
  text-align: center;
  cursor: pointer;
  border-radius: 15px;
  color: rgb(230, 230, 230);
}

.settings .settings-close:hover {
  color: white;
}

.settings .settings-close svg {
  background: var(--electron-background-2);
  box-shadow: 0 0 12px 6px var(--electron-background-2);
}