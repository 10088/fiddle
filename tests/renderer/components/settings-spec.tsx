import { shallow } from 'enzyme';
import * as React from 'react';

import { Settings } from '../../../src/renderer/components/settings';

jest.mock('../../../src/renderer/components/settings-github', () => ({
  GitHubSettings: 'settings-github'
}));

jest.mock('../../../src/renderer/components/settings-electron', () => ({
  ElectronSettings: 'settings-electron'
}));

jest.mock('../../../src/renderer/components/settings-credits', () => ({
  CreditsSettings: 'settings-credits'
}));

describe('CreditsSettings component', () => {
  beforeEach(() => {
    this.store = {
      isSettingsShowing: true
    };
  });

  it('renders null if settings not showing', () => {
    this.store.isSettingsShowing = false;

    const wrapper = shallow(
      <Settings appState={this.store} />
    );

    expect(wrapper.html()).toBe(null);
  });

  it('renders only the menu if page unknown', () => {
    const wrapper = shallow(
      <Settings appState={this.store} />
    );

    wrapper.setState({ section: 'blub' });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the Electron page by default', () => {
    const wrapper = shallow(
      <Settings appState={this.store} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders the GitHub page after a click', () => {
    const wrapper = shallow(
      <Settings appState={this.store} />
    );

    wrapper.find('.GitHub').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the GitHub page after a click', () => {
    const wrapper = shallow(
      <Settings appState={this.store} />
    );

    wrapper.find('.Credits').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
