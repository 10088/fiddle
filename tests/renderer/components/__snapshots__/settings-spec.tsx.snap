// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditsSettings component renders only the menu if page unknown 1`] = `
<div
  className="settings"
>
  <div
    className="settings-menu"
  >
    <ul>
      <li
        className="General"
        key="General"
        onClick={[Function]}
      >
        General
      </li>
      <li
        className="Electron"
        key="Electron"
        onClick={[Function]}
      >
        Electron
      </li>
      <li
        className="Credits"
        key="Credits"
        onClick={[Function]}
      >
        Credits
      </li>
    </ul>
  </div>
  <div
    className="settings-content"
  >
    <div
      className="settings-close"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f057",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
            ],
            "iconName": "times-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
    </div>
  </div>
</div>
`;

exports[`CreditsSettings component renders the Electron page by default 1`] = `
<div
  className="settings"
>
  <div
    className="settings-menu"
  >
    <ul>
      <li
        className="selected General"
        key="General"
        onClick={[Function]}
      >
        General
      </li>
      <li
        className="Electron"
        key="Electron"
        onClick={[Function]}
      >
        Electron
      </li>
      <li
        className="Credits"
        key="Credits"
        onClick={[Function]}
      >
        Credits
      </li>
    </ul>
  </div>
  <div
    className="settings-content"
  >
    <div
      className="settings-close"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f057",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
            ],
            "iconName": "times-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
    </div>
    <settings-general
      appState={
        Object {
          "isSettingsShowing": true,
        }
      }
    />
  </div>
</div>
`;

exports[`CreditsSettings component renders the General page after a click 1`] = `
<div
  className="settings"
>
  <div
    className="settings-menu"
  >
    <ul>
      <li
        className="selected General"
        key="General"
        onClick={[Function]}
      >
        General
      </li>
      <li
        className="Electron"
        key="Electron"
        onClick={[Function]}
      >
        Electron
      </li>
      <li
        className="Credits"
        key="Credits"
        onClick={[Function]}
      >
        Credits
      </li>
    </ul>
  </div>
  <div
    className="settings-content"
  >
    <div
      className="settings-close"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f057",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
            ],
            "iconName": "times-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
    </div>
    <settings-general
      appState={
        Object {
          "isSettingsShowing": true,
        }
      }
    />
  </div>
</div>
`;

exports[`CreditsSettings component renders the General page after a click 2`] = `
<div
  className="settings"
>
  <div
    className="settings-menu"
  >
    <ul>
      <li
        className="General"
        key="General"
        onClick={[Function]}
      >
        General
      </li>
      <li
        className="Electron"
        key="Electron"
        onClick={[Function]}
      >
        Electron
      </li>
      <li
        className="selected Credits"
        key="Credits"
        onClick={[Function]}
      >
        Credits
      </li>
    </ul>
  </div>
  <div
    className="settings-content"
  >
    <div
      className="settings-close"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f057",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
            ],
            "iconName": "times-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
    </div>
    <settings-credits
      appState={
        Object {
          "isSettingsShowing": true,
        }
      }
    />
  </div>
</div>
`;
